### PR TITLE
imp(all): Move latest changes from Evmos main (incl. SDK v50)

### PR DESCRIPTION
**Pull request by MalteHerrmann**
This PR moves the latest changes from the https://github.com/evmos/evmos `main` branch into the OS repository. This includes a bunch of dependency bumps and other refactors, most notably bumping the Cosmos SDK version to v0.50.9.

## Summary by CodeRabbit

- **New Features**
	- Introduced a benchmarking suite for performance testing of various transaction types in the Evmos blockchain.
	- Added an integration test suite for validating Cosmos transaction execution and reward management scenarios.

- **Improvements**
	- Streamlined transaction creation and signing processes, enhancing clarity and maintainability.
	- Enhanced account management capabilities within the test suite.

- **Bug Fixes**
	- Improved accuracy in gas wanted calculations across multiple transaction types. 

- **Documentation**
	- Updated testing framework to ensure comprehensive coverage for transaction handling and performance analysis.

### CODE REVIEW

**Comment by 0xstepit** [comment]
@MalteHerrmann we should add some branch protection rule because ATM I could squash and merge without any approval of the PR.

**Review by 0xstepit — APPROVED** [review]
Great!

File: api/os/evm/v1/msg.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		return nil, fmt.Errorf("invalid type, expected MsgEthereumTx and got %T", msg)
+	}
+
+	txData, found := supportedTxs[msgEthTx.Data.TypeUrl]
+	if !found {
+		return nil, fmt.Errorf("invalid TypeUrl %s", msgEthTx.Data.TypeUrl)
+	}
+
+	// msgEthTx.Data is a message (DynamicFeeTx, LegacyTx or AccessListTx)
+	if err := msgEthTx.Data.UnmarshalTo(txData); err != nil {
```
[the remark this answers is not in the document]
Comment (MalteHerrmann) [reply api/os/evm/v1/msg.go:46]: @evmos/core-engineering what do you guys think of this, should we adjust the handling here for the `GetSigners`? I mean the proposed change looks good to me, not sure if this concurrency problem is an actual thing though 👀

**Comment by hanchon** [comment]
> @MalteHerrmann we should add some branch protection rule because ATM I could squash and merge without any approval of the PR.

I'll update the configuration tomorrow morning!

File: api/os/evm/v1/msg.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (GAtom22) [reply api/os/evm/v1/msg.go:46]: yeah, I we can go with this solution. Is simple and we discard the possibility if this having side effects

[the remark this answers is not in the document]
Comment (GAtom22) [reply api/os/evm/v1/msg.go:46]: adding these changes on the evmos repo here https://github.com/evmos/evmos/pull/2884

**Comment by hanchon** [comment]
LGTM! 🚀

**Review by hanchon — APPROVED** [review]

**Pull request merged**